### DISCUSSION TITLE
Is it possible to add variants to an element that doesn't declare any?

### DISCUSSION THREAD

**Discussion by michaellenaghan** [jmjuanes/siimple#197] [Q&A] [answered]
I would like to add a `readable` variant to `container`, like this:

```js
  readable: {
    maxWidth: "50rem",
  },
```

so that I can do this:

```html
  <div class="container is-readable">
    ...
  </div>
```

I've seen how `preset-ice` and `preset-mustard` declare variants for `alerts`, `badges` and `buttons`. But they can do that because the `alert`, `badge` and `button` elements declare those kinds of variants in advance.

Is it possible to add variants to an element that doesn't declare any in advance?

(Of course, it would be easy to just add the class. I kind of see this as an opportunity to learn something about how the framework works, though.)

**Comment by jmjuanes** [comment, marked as answer]
You are right, only `alert`, `badge`, `button` and `spinner` can generate additional variants automatically using the extra keys defined in its configuration (see the table in https://siimple.xyz/elements).

Instead of just adding the class, you can use the `layout.container` field of the configuration to extend the `container` element with the styles and modifiers that you need. Your example can be easily generated with the following configuration:

```js
export default {
    // ...other configuration
    layout: {
        container: {
            // ...other container styles
            "&.is-readable": {
                maxWidth: "50rem",
            },
        },
    },
};
```

The `&` parent selector works like in SASS (https://sass-lang.com/documentation/style-rules/parent-selector), so in this case will select the container class.

I have created a [playground with this example](https://siimple.xyz/playground#html=DwEwlgbgBAxgNgQwM5ILwCIYHsB2AXBMHAUwCcowkBaU4hEBAIzmPQD4AoKbqAYVzzF8HYAHpwETkA%3D%3D&config=JYWwDg9gTgLgBAIwIYGcCmcBmUIjgIgAEVhQwAbNAejCjXRgFpl18BuAKDQA9JY4AJmkxIAruXgBvDnFlwQEAePoAuOJIC%2BAGhlzR6AELQhUI9zUwootDrlx9aAEoQIMAMowAnpRQWrN3VkAOhCWALtyJE8IURg1aTs7AGMIADsYJGBUtCh4wMS7fAAyIOAURjokASQESnw8gsa5Kip5JG4AdWABGAALNXwAVigABjQQfFsmxuQkgGsAcxxRVIEAYQhyaAGAYhF9pEn86bkQdq6e%2FoIAZhG6CamT7WO5Z7tnjU4gA%3D%3D%3D).

**Reply by michaellenaghan** [reply]
Ah, interesting, thank you!
